app/templates/browse.hbs

```diff
@@ -33,9 +33,9 @@
             </div>
             <ul class="list subcategories hide-for-small">
               {{#each category.childCategories as |childCategory|}}
-                {{#link-to 'package_category' childCategory.id class="custom_a"}}
+                <div class="custom_a" {{action "viewChildCategoryItems" childCategory}}>
```

Comment:
You can try `link-to` tag along with action to remove `viewChildCategoryItems`.
```suggestion
     {{#link-to 'package_category' childCategory.id class="custom_a"}}
        <div {{action 'setChildCategory' target="packageCategory">
```
And inside `packageCategory` controller add one method `setChildCategory` which will have this set action.
```
 actions: {
    setChildCategory(category) {
       this.set("selectedCategoryId", category)
   }
 }
```

Reply:
would both of them trigger ? 🤔 

I think `setChildCategory` can also be removed by using the `mut` helper 
https://medium.com/@_edhuang/ember-mut-helper-lets-mutate-acdf6b58a0df

I've never used it though, so not 100% sure. I think @Darshan-Chauhan might know

Reply:
Thanks for the suggestion @abulsayyad123 , but this wont work. Will break it. and the functionality which is required will not work with this code.

Reply:
`{{#link-to 'package_category' childCategory.id}}`
This will generate url as `/category/{{childCategoryId}}` which is not the requirement. We should be able to switch between categories from the parent url without changing the URL.
Thanks

Reply:
I can change the method name if you say so, Will rename it to `setChildCategory`.
Do let me know in case.
Thanks @abulsayyad123

Reply:
using `eventpropagation` it can be avoided. My suggestion was considering repetition of same logic twice in controller. So, i tried it to handle it within templates itself.